utils/lewis_launcher.py

```diff
@@ -185,27 +200,49 @@ def backdoor_set_on_device(self, variable_name, value):
         else:
             self.backdoor_command(["device", str(variable_name), str(value)])
 
+    def backdoor_run_function_on_device(self, function_name, arguments=None):
+        """
+        Run a function in lewis using the back door on a device.
+
+        :param function_name: name of th function to call
+        :param arguments: an iterable of the arguments for the function; None means no arguments. Arguments will
+            automatically be turned into json
+        :return:
+        """
+        command = ["device", function_name]
+        if arguments is not None:
+            for argument in arguments:
```

Comment:
I prefer not to repeat `command.append` in case of change of type. You could rather do:

```
formatted_arg = "'{}'".format(argument) if isinstance(argument, str) else str(argument)
command.append(formatted_argument)
```

or you could go all the way and do:

`command += ["'{}'".format(a) if isinstance(a, str) else str(a) for a in arguments]`

That is starting to sacrifice readability though